### ISSUE TITLE
Allow multiple Glue database datasets creation in the same environment

### ISSUE THREAD

**Issue by diandramelo** [awslabs/aws-dataall#306]
Hello there!

I would like to be able to **add multiple tables from different databases in Glue** into the data.all interface. I was able to succesfully create my first dataset, and all tables within the Glue database where added to this dataset.

In a same data.all environment (AWS account), I have **multiple Glue databases** and I would like to add all of their tables into datasets, creating one dataset per database. The problem is once I have already created the first stack of Glue dataset in an environment, the Glue job created (**DatasetGlueProfilingJob**) has not an unique name, and for new stacks under the same AWS environment the pipeline will not be executed because of an already existant job Glue. 

Here's how the error appears in CloudFormation:

> datalake-platform-prod-profiler already exists in stack arn:aws:cloudformation:eu-west-1:XXXXXXXXXXX:stack/dataall-dataset-h7y1um8l/6d0ac1b0-a88d-11ed-8846-0637c286448b

Does that mean that I would have to create multiple environments for one AWS account in order to have multiple DatasetGlueProfilingJob for each database?

Please let me know if it is still not clear, I would be glad to explain it!

Best regards,

Diandra MELO.

**Comment by dlpzx** [comment]
Hi @diandramelo! First, I edited your comment because it was showing the Account Id. If this was a production account you will need to evaluate the security risk. 

I am checking your issue, which I think it comes from the data.all Dataset definition, not from the Environment.For each new data.all Dataset, data.all deploys a CloudFormation stack `arn:aws:cloudformation:eu-west-1:<AWS_ACCOUNT>:stack/dataall-dataset-u3x7oxf6/...` 

This stack includes the Glue profiling job that you mention:
![image](https://user-images.githubusercontent.com/71252798/218796693-ea4f5eff-f524-4a86-8e97-5a2565db6e4e.png)

I checked the code, the glue profiling job name is defined in `https://github.com/awslabs/aws-dataall/blob/main/backend/dataall/db/api/dataset.py` line 175 `dataset.GlueProfilingJobName = f'{dataset.S3BucketName}-profiler'`

So I think I found your issue, in data.all one Dataset = one Glue Database + one S3 Bucket. I think you are trying to create multiple Datasets = multiple Glue Databases + the same S3 Bucket.  Since the bucketName is the same one it cannot create more profiling jobs, crawlers.. for that bucket. You would need to create a dedicated S3 Bucket for each database.

In conclusion, it is not a bug, but an expected behavior. However, this can turn into a feature request if you have the need to create one Dataset = one database and shared S3 Bucket. We would need to see the implications of this change but first we need to understand why do you need this and how you would like it to work.

You can create a new issue or we can work on this one :)

**Comment by diandramelo** [comment]
Hello @dlpzx,

First of all, I would like to thank you for your quick response and for editing my comment! I haven't thought of the account ID on the error message...

Your answer is pretty clear and answers my question. I do understand the need of having one S3 Bucket per dataset. 

I started to construct a datalake in AWS environment with a layered structure (bronze / silver / gold), having one bucket per layer as originary buckets. I use separated S3 buckets for each database but they all belong to the same originary bucket, and since we cannot work with S3 URIs in data.all dataset ingestion this gave me a conflict error of same "main" S3 Bucket.

I will review a new structure with one main bucket per database, and in case when this does not fit the needs of the company I might open a feature request. Thank you again!
